lib/mastermind/score_filter_ai.rb

```diff
@@ -23,8 +19,12 @@ def next_guess(feedback)
 
     attr_reader :initial_guess, :set_of_guesses, :current_guess, :code_score_calc 
 
-    def update_set_of_guesses(feedback)
-      @set_of_guesses = @set_of_guesses.select do |a_guess|
+    def initial_guess?(feedback)
+      feedback == ScoringFeedback.new(-1, -1)
+    end
+
+    def filter_set_of_guesses(set_of_guesses, feedback)
+      set_of_guesses.select do |a_guess|
         feedback == code_score_calc.score(current_guess, a_guess)
       end
     end
```

Comment:
I dislike the magic variables but that's a trade off here. Are these -1 values duplicated elsewhere? If so, I'd reject this change. If it's all in one place, i.e., if you change it, you change it in one place, then I'd be inclined to say this is an improvement.